Net/src/SocketImpl.cpp

```diff
@@ -1433,21 +1434,26 @@ Poco::Int64 _sendfile(poco_socket_t sd, FileIOS::NativeHandle fd, Poco::UInt64 o
 	return sent;
 }
 
-Poco::Int64 SocketImpl::sendFile(FileInputStream &fileInputStream, Poco::UInt64 offset)
+IntPtr SocketImpl::sendFile(FileInputStream &fileInputStream, UIntPtr offset)
 {
 	FileIOS::NativeHandle fd = fileInputStream.nativeHandle();
-	Poco::UInt64 fileSize = fileInputStream.size();
+	UIntPtr fileSize = fileInputStream.size();
 	std::streamoff sentSize = fileSize - offset;
-	Poco::Int64 sent = 0;
+	IntPtr sent = 0;
 	sighandler_t sigPrev = signal(SIGPIPE, SIG_IGN);
 	while (sent == 0)
 	{
 		errno = 0;
 		sent = _sendfile(_sockfd, fd, offset, sentSize);
+		if (sent < 0)
+		{
+			error(errno, std::string("[sendfile error]") + Error::getMessage(errno));
```

Comment:
Does not seem right. See other places where error() is used It should not use Error::getMessage(errno) as it looks like error will actually check hose. But anyway. This is probably even not the right place to do error handling. Every function should do throwing at they own. So _sendFile should be place to throw

Reply:
_sendFile is a free function it can't throw exception by call error, code will be too complex
and _sendFile uses for *nix only, on windows doesn't